### PR TITLE
add test case to check the existence of "books this page is in" list

### PR DESCRIPTION
**Pull request by BrianYing**
#220

### CODE REVIEW

File: tests/utils.py [hunk 1]
```diff
@@ -71,3 +73,33 @@ def skip_if_destructive_and_sensitive(request, base_url):
...
+        raise IndexError('Unknown dimension: %s' % dimension)
+    return get_size[dimension]
+
+
+def set_window_size(self, width=0, height=0):
+    """Attempt to change the browser window size."""
+    if width >= 1 and height >= 1:
+        self.driver.set_window_size(width, height)
+        sleep(1.0)
+    # return self.get_window_size()
```
Comment (m1yag1) [inline tests/utils.py]: Do not merge code with commented code. If there is commented code it either needs to be deleted or be necessary.

File: tests/utils.py [hunk 2]
```diff
@@ -71,3 +73,33 @@ def skip_if_destructive_and_sensitive(request, base_url):
 def shorten_tag(tag):
     """Returns the short version of a git tag when given the long (or short) version."""
     return tag.split('-')[0]
+
+
+def get_window_size(self, dimension=None):
```
Comment (m1yag1) [inline tests/utils.py]: functions outside of a class with a reference to self are not good. You'll need to place these within a class. You could add them to the base page and they would be available everywhere.

File: tests/webview/ui/test_content.py [hunk 3]
```diff
@@ -784,3 +786,35 @@ def test_book_containing_title_not_limited(webview_base_url, selenium, page_id):
...
+    # GIVEN the webview base url, page_id, and the Selenium driver
+
+    # WHEN we visit that page of the chapter and we have a list of books containing the page
+    content = ContentPage(selenium, webview_base_url, id=page_id).open()
+
+    # THEN check if the books list exists and on the left
+    assert content.books_containing.book_list
+
+    window_width = utils.get_window_size(content, 'width')
+    assert content.location['x'] < window_width/2
```
Comment (m1yag1) [inline tests/webview/ui/test_content.py]: would `selenium.get_window_size()['width'] / 2`  not work. Additionally, you should be placing a space around operators in an expression such as in my example.

File: pages/webview/base.py [hunk 4]
```diff
@@ -74,6 +76,31 @@ def offscreen_click(self, element):
...
+            raise IndexError('Unknown dimension: %s' % dimension)
+        return get_size[dimension]
+
+    def set_window_size(self, width=0, height=0):
+        """Attempt to change the browser window size."""
+        if width >= 1 and height >= 1:
+            self.driver.set_window_size(width, height)
+            sleep(1.0)
+
+    def set_window_position(self, x_=0, y_=0):
```
Comment (m1yag1) [inline pages/webview/base.py]: Why `x_` instead of `x`?

File: tests/webview/ui/test_content.py [hunk 5]
```diff
@@ -816,5 +814,5 @@ def test_books_this_page_in_list(webview_base_url, selenium, page_id):
     # THEN check if the books list exists and on the left
     assert content.books_containing.book_list
 
-    window_width = utils.get_window_size(content, 'width')
-    assert content.location['x'] < window_width/2
+    window_width = content.get_window_size('width')
```
Comment (m1yag1) [inline tests/webview/ui/test_content.py]: move to under the `WHEN` clause and edit WHEN comment.

`# WHEN we load the page of the chapter and we have the width of the window`

File: pages/webview/content_page.py [hunk 6]
```diff
@@ -11,14 +13,26 @@ def loaded(self):
...
+    @property
+    def location(self):
+        content = self.find_element(By.CSS_SELECTOR, ".table-of-contents")
+        return content.location
+
     class BooksContaining(Region):
         _root_locator = (By.CLASS_NAME, 'booksContaining')
         _book_list_locator = (By.CSS_SELECTOR, 'div > ul > li')
 
+        # @property
```
Comment (m1yag1) [inline pages/webview/content_page.py]: Don't leave comments in code you want merged.

File: tests/webview/ui/test_content.py [hunk 7]
```diff
@@ -784,3 +784,35 @@ def test_book_containing_title_not_limited(webview_base_url, selenium, page_id):
...
+
+
+@markers.webview
+@markers.test_case('C195054')
+@markers.nondestructive
+@markers.parametrize('page_id', ['4fGVMb7P@1'])
+def test_books_this_page_in_list(webview_base_url, selenium, page_id):
+    # GIVEN the webview base url, page_id, and the Selenium driver
+
+    # WHEN we visit that page of the chapter and we have a list of books containing the page
```
Comment (m1yag1) [inline tests/webview/ui/test_content.py]: Comment is not exactly accurate... you don't have a list of books containing the page.

File: pages/webview/base.py (hunk 4 above)
Comment (BrianYing) [reply pages/webview/base.py]: Honestly, I don't know. This are some old codes that @gregfitch wrote for automation and I just copied.

File: tests/webview/ui/test_content.py [hunk 8]
```diff
@@ -784,3 +784,34 @@ def test_book_containing_title_not_limited(webview_base_url, selenium, page_id):
...
+    # THEN the button name is "Books" instead of "Contents"
+    btn_name = content.header_nav.contents_button.text
+    assert btn_name == "Books"
+
+
+@markers.webview
+@markers.test_case('C195054')
+@markers.nondestructive
+@markers.parametrize('page_id', ['4fGVMb7P@1'])
+def test_books_this_page_in_list(webview_base_url, selenium, page_id):
```
Comment (m1yag1) [inline tests/webview/ui/test_content.py]: I missed this last time. This test needs to be renamed to something like `test_books_containing_list_is_on_left_of_page`

**Pull request merged**